### PR TITLE
Implementation of ArgMin ONNX + TF extractors

### PR DESCRIPTION
**Pull request by yekruglov**
Description:
* Added ArgMin internal operation and ArgMin extractors
* ArgMax infer function is used for ArgMin operation since it contains only shape inference. Renamed it to `arg_ops_infer`
* Reused `ArgMaxToTopK` transformation for ArgMin op. Renamed it to `ArgOpsToTopK`.  Same for `ArgMaxSqueeze`.
* ArgMax operation has an attribute `out_max_val`. As I understand, this attribute is valid only for Caffe ArgMax, so for TF and ONNX it is equal to 0. I do not see any reasons to add an attribute `out_min_val` for MO ArgMin operation, because for TF and ONNX it is useless and I have not found any mentions of this operation in Caffe.

Ticket: 51639

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests
* [ ]  Framework operation tests
* [x]  Transformation tests
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list
* [x]  Guide on how to convert the **public** model N/A: no new models were enabled
* [x]  User guide update N/A

### CODE REVIEW

**Comment by rblaczkowski** [comment]
Please merge current master to your branch to avoid OpenVINO ONNX CI issues.

**Review by popovaan — APPROVED** [review]

**Review by mvafin — APPROVED** [review]

**Comment by lazarevevgeny** [comment]
@yekruglov , please, rebase with the latest master.

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: model-optimizer/extensions/front/tf/argmin_ext.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    @classmethod
+    def extract(cls, node):
+        attrs = {
+            'top_k': 1,
+            'axis': None,
+            'keepdims': 0,
+            'remove_values_output': True,
+            'output_type': tf_dtype_extractor(node.pb.attr['out_type'].type, np.int64)
+        }
+        ArgMaxOp.update_node_stat(node, attrs)
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/front/tf/argmin_ext.py:24]: Should be `ArgMinOp`

File: model-optimizer/extensions/front/tf/argmin_ext.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    enabled = True
+
+    @classmethod
+    def extract(cls, node):
+        attrs = {
+            'top_k': 1,
+            'axis': None,
+            'keepdims': 0,
+            'remove_values_output': True,
+            'output_type': tf_dtype_extractor(node.pb.attr['out_type'].type, np.int64)
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/front/tf/argmin_ext.py:22]: The TF operation attribute name is `output_type` not `out_type`. The same comment for the `ArgMax` operation

File: model-optimizer/extensions/middle/ArgOpsToTopK.py [hunk 3]
```diff
@@ -36,7 +36,9 @@ def replace_pattern(self, graph: Graph, match: dict):
 
         assert axis is not None, 'The "axis" should be defined for node "{}"'.format(node_name)
         assert node.has_and_set('output_type'), 'The data type is not set for node "{}"'.format(node_name)
-        topk_node = TopK(graph, {'axis': axis, 'mode': 'max', 'sort': 'index',
+
+        topk_mode = 'max' if node['op'] == 'ArgMax' else 'min'
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/middle/ArgOpsToTopK.py:40]: `node['op']` -> `node.op`

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

**Review by iimironov — APPROVED** [review]

File: model-optimizer/extensions/ops/argmax.py [hunk 4]
```diff
@@ -10,14 +10,49 @@
...
+        out_shape = np.array(shape)
+        out_shape[axis] = node.top_k
+        PermuteAttrs.create_permute_attrs(node, attrs=[('axis', 'input:0')])
+    else:
+        out_shape[0] = shape[0]
+        out_shape[2] = node.top_k
+        if node.has_and_set('out_max_val'):
+            out_shape[1] = 2
+
+    node.out_node().shape = out_shape
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/ops/argmax.py:45]: node.out_port(0).data.set_shape(out_shape)

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: model-optimizer/extensions/ops/argmax.py [hunk 5]
```diff
@@ -10,14 +10,49 @@
...
+
+
 class ArgMaxOp(Op):
     op = 'ArgMax'
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/ops/argmax.py:50]: Add `enabled = False`

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**